src/main/java/com/thebois/models/world/structures/Structure.java

```diff
@@ -0,0 +1,63 @@
+package com.thebois.models.world.structures;
+
+import java.util.Objects;
+
+import com.thebois.models.Position;
+
+/**
+ * Base structure for the game.
+ */
+public class Structure implements IStructure {
```

Comment:
We will never have just a structure placed in the world so this class could be made abstract imo. And have a House class implement it.

Reply:
Since there is no difference between a house and a structure it doesn't really matter, but it would make sense to make structure abstract for the future. Will fix.